src/regmod/data.py

```diff
@@ -11,7 +11,85 @@
 
 @dataclass
 class Data:
-    col_obs: Union[str, List[str]] = None
+    """Data class used for validate and accessing data in `pd.DataFrame`.
+
+    Parameters
+    ----------
+    col_obs : Optional[Union[str, List[str]]], optional
+        Column name(s) for observation. Default to be `None`.
+    col_covs : List[str], optional
+        Column names for covariates. Default to be an empty list.
+    col_weights : str, default="weights"
```

Comment:
Not sure if you should list the default where the type is specified, or just in the description